cmd/kubeadm/app/util/tokens.go

```diff
@@ -20,65 +20,74 @@ import (
 	"crypto/rand"
 	"encoding/hex"
 	"fmt"
+	"regexp"
+	"strconv"
 	"strings"
 
 	kubeadmapi "k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm"
+	kubeadmapiext "k8s.io/kubernetes/cmd/kubeadm/app/apis/kubeadm/v1alpha1"
 )
 
 const (
-	TokenIDLen = 6
-	TokenBytes = 8
+	TokenIDBytes = 3
+	TokenBytes   = 8
 )
 
-func RandBytes(length int) ([]byte, string, error) {
+func RandBytes(length int) (string, error) {
 	b := make([]byte, length)
 	_, err := rand.Read(b)
 	if err != nil {
-		return nil, "", err
+		return "", err
 	}
-	// It's only the tokenID that doesn't care about raw byte slice,
-	// so we just encoded it in place and ignore bytes slice where we
-	// do not want it
-	return b, hex.EncodeToString(b), nil
+	return hex.EncodeToString(b), nil
 }
 
-func GenerateToken(s *kubeadmapi.Secrets) error {
-	_, tokenID, err := RandBytes(TokenIDLen / 2)
+func GenerateToken(d *kubeadmapi.TokenDiscovery) error {
```

Comment:
I'm not saying you have to do it now, but can you move this file to a separate package for clarity (`cmd/kubeadm/app/util/token`) and import it as `tokenutil` where it's then used?